tests/src/JIT/SIMD/Vector3Interop.cs

```diff
@@ -0,0 +1,339 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Numerics;
+using System.Runtime.InteropServices;
+using System.Text;
+
+
+public struct DT 
+{
+    public Vector3 a;
+    public Vector3 b;
+};
+
+[StructLayout(LayoutKind.Sequential, CharSet = CharSet.Ansi)]
+public struct ComplexDT 
+{
+    public int iv;
+    public DT vecs;
+    [MarshalAs(UnmanagedType.ByValTStr, SizeConst=256)] public string str;
+    public Vector3 v3;
+};
+
+class PInvokeTest 
+{
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]
+    public static extern int nativeCall_PInvoke_CheckVector3Size();
+
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]
+    public static extern float nativeCall_PInvoke_Vector3Arg(
+        int i, 
+        Vector3 v1, 
+        [MarshalAs(UnmanagedType.LPStr)] string s, 
+        Vector3 v2);
+
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]
+    public static extern Vector3 nativeCall_PInvoke_Vector3Ret();
+
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]
+    public static extern float nativeCall_PInvoke_Vector3Array(Vector3[] v_array);
+
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]
+    public static extern DT nativeCall_PInvoke_Vector3InStruct(DT d);
+
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]
+    public static extern void nativeCall_PInvoke_Vector3InComplexStruct(ref ComplexDT cdt);
+    
+    public static bool test() 
+    {
+
+        // Expected return value is 12 bytes.
+        if (nativeCall_PInvoke_CheckVector3Size() != 12) 
+        {
+            Console.WriteLine("The size of native Vector3 type is not 12 bytes");
+            return false;            
+        }
+    
+        {
+            int iv = 123;
+            Vector3 v1 = new Vector3(1,2,3);
+            string str = "abcdefg";
+            Vector3 v2 = new Vector3(10,11,12);
+            // Expected return value = 1 + 2 + 3 + 10 + 11 + 12 = 39
+            if (nativeCall_PInvoke_Vector3Arg(iv, v1, str, v2) != 39) 
+            {
+                Console.Write("PInvoke Vector3Arg test failed\n");
+                return false;            
+            }
+        }
+        
+        // JIT crashes with this testcase.
+        // Disabled temporarily.
+        // {
+        //     // Expected return value = 1 + 2 + 3 = 6
+        //     Vector3 ret = nativeCall_PInvoke_Vector3Ret();
```

Comment:
When we enable this test, please also update this test to compute DotProduct(ret, ret).